lua/vex_library/modules/server/e2helperfuncs.lua

```diff
@@ -16,20 +17,127 @@ vex.registerExtension = function(name,enabled,helpstr,...)
     E2Lib.RegisterExtension(name,enabled,helpstr,...)
 end
 
--- Note: This is terrible
-vex.getE2Type = function(val)
-    local e2types = wire_expression_types
-    for TypeName,TypeData in pairs(e2types) do
-        if type(val) == "table" then
-            -- These are incredibly hacky but they work
-            if val.size then return "TABLE" end
-                if #val == 3 and isnumber(val[1]) and isnumber(val[2]) and isnumber(val[3]) then return "VECTOR" end
-                return "ARRAY"
+vex.registerConstant = function(name, value)
+    -- This exists because E2Lib doesn't return a passed value, which can be handy (see selfaware2).
+    E2Lib.registerConstant(string.upper(name), value) -- Upper shouldn't be used a lot
```

Comment:
There's no need to call `string.upper` here, the E2Lib will do that.

Reply:
https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/init.lua#L166